server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -77,6 +77,14 @@ const orderMutations = {
       const collective = await loadAccount(order.toAccount);
       const paymentMethod = await getLegacyPaymentMethodFromPaymentMethodInput(order.paymentMethod);
 
+      /* Currently we only support the Giving Block related values in the data object
+       *  TODO: Maybe in future we could extend this to all existing data fields
+       */
+      let orderData;
+      if (order.data) {
+        orderData = pick(order.data, ['thegivingblock']);
```

Comment:
We should move this `pick` to the common `createOrder` mutation, as we always want this sanitization to happen. Also, `pick` is powerful enough for us so sanitize with more granularity:

```suggestion
        orderData = pick(order.data, ['thegivingblock.pledgeCurrency', 'thegivingblock.pledgeAmount']);
```

Could be worth extracting this list to a shared `ORDER_PUBLIC_DATA_FIELDS` (or similar) since we're using it below.